tests/unit/nupic/research/tp_test.py

```diff
@@ -184,6 +187,59 @@ def testCheckpointMiddleOfSequence2(self):
     self.assertTPsEqual(tp2, tp4)
 
 
+  def testWriteRead(self):
+    tp1 = TP(2048, 32, 0.21, 0.5, 11, 20, 0.1, 0.1, 1.0, 0.0, 14, False, 5, 2,
+             True, 1960, 0, False, 3, 10, 5, 0, 32, 128, 32, 'normal')
+    tp1.collectSequenceStats = True
+    tp1.resetStats()
+
+    with open(resource_filename(__name__, 'data/tp_input.csv'), 'r') as fin:
+      reader = csv.reader(fin)
+      records = []
+      for bottomUpInStr in fin:
+        bottomUpIn = numpy.array(eval('[' + bottomUpInStr.strip() + ']'),
+                                 dtype='int32')
+        records.append(bottomUpIn)
+
+    i = 1
+    for r in records[:250]:
```

Comment:
I decreased from 250 to 150 (SegmentUpdates are created after more than 100 iterations) and after deserialization I decreased the number from 250 to 10 iterations to compare outputs.